### PR TITLE
fix: Use appropriate aria-haspopup for select/multiselect with filtering

### PR DESCRIPTION
**Pull request by gethinwebster**
### Description

Uses `aria-haspopup="dialog"` for select & multiselect with filtering,
as the previous value of "listbox" is not appropriate for something that's
more complex than a simple list.

Also adjust roles and `aria-controls` to align.

Related links, issue #, if available: AWSUI-21334

### How has this been tested?

Unit tests, and tested with VO & Chrome

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/dropdown/interfaces.ts [hunk 1]
```diff
@@ -132,6 +132,15 @@ export interface DropdownProps extends ExpandToViewport {
...
   onBlur?: NonCancelableEventHandler<Pick<React.FocusEvent, 'target' | 'relatedTarget'>>;
+
+  /**
+   * ID for the dropdown content wrapper
+   */
+  dropdownContentId?: string;
+  /**
+   * HTML role for the dropdown content wrapper
+   */
+  dropdownContentRole?: string;
```
Comment (timogasda) [inline src/internal/components/dropdown/interfaces.ts:143]: Would it make sense to "enum" this like we do `ariaHasPopup`?

Comment (gethinwebster) [reply src/internal/components/dropdown/interfaces.ts:143]: Given that there are so many options I'm not sure it makes so much sense here. We could _perhaps_ limit it to the options we think might be relevant here, but especially as it's an internal interface I'm not sure it has so much value

Comment (timogasda) [reply src/internal/components/dropdown/interfaces.ts:143]: Right, but `ariaHasPopup` is an internal prop as well, right?

And yes, I would limit it to the values that we currently think are reasonable (which I think is just `dialog`?).

Not a blocking comment, though. I'll leave the final call up to you.

**Review by timogasda — APPROVED** [review]
One non-blocking comment

**Pull request merged**